scipy/interpolate/fitpack.py

```diff
@@ -559,68 +331,54 @@ def splev(x, tck, der=0, ext=0):
     -------
     y : ndarray or list of ndarrays
         An array of values representing the spline function evaluated at
-        the points in ``x``.  If `tck` was returned from `splprep`, then this
+        the points in `x`.  If `tck` was returned from `splprep`, then this
         is a list of arrays representing the curve in N-dimensional space.
 
+    Notes
+    -----
+    Manipulating the tck-tuples directly is not recommended. In new code,
+    prefer using the `BSpline` objects.
+
     See Also
     --------
     splprep, splrep, sproot, spalde, splint
     bisplrep, bisplev
+    BSpline
 
     References
     ----------
     .. [1] C. de Boor, "On calculating with b-splines", J. Approximation
         Theory, 6, p.50-62, 1972.
-    .. [2] M.G. Cox, "The numerical evaluation of b-splines", J. Inst. Maths
+    .. [2] M. G. Cox, "The numerical evaluation of b-splines", J. Inst. Maths
         Applics, 10, p.134-149, 1972.
     .. [3] P. Dierckx, "Curve and surface fitting with splines", Monographs
         on Numerical Analysis, Oxford University Press, 1993.
 
     """
-    t, c, k = tck
-    try:
-        c[0][0]
-        parametric = True
-    except:
-        parametric = False
-    if parametric:
-        return list(map(lambda c, x=x, t=t, k=k, der=der:
-                        splev(x, [t, c, k], der, ext), c))
+    if isinstance(tck, BSpline):
+        t, c, k = tck
+
+        # fitpack interpolates along the last axis
```

Comment:
I would actually suggest a DeprecationWarning here --- the user should call `bspline(x)` instead of getting the sub-optimal `list(map(... ))` Python-side workaround.

The sub-optimal behavior is OK if tuples passed in, however.